brayns/engine/scene/ModelsOperations.h

```diff
@@ -20,42 +20,38 @@
 
 #pragma once
 
-#include <brayns/engine/geometry/GeometryView.h>
-#include <brayns/engine/model/Model.h>
-#include <brayns/engine/model/ModelComponents.h>
+#include "ModelManager.h"
 
 namespace brayns
 {
-/**
- * @brief Adds clipping geometry to the model (Affects the whole scene)
- */
-class ClippingComponent : public Component
+class ModelsOperations
 {
 public:
-    template<typename T>
-    ClippingComponent(T primitive)
-        : ClippingComponent(std::vector<T>{std::move(primitive)})
+    template<typename Component>
+    static void removeModelsWithComponent(ModelManager &models)
     {
-    }
+        static_assert(std::is_same_v<Component, std::decay_t<Component>>, "Only decayed types allowed");
```

Comment:
This check should be at the model level (~component manager) no ?

Otherwise it should be made everywhere we use components no ?

If you need to make this check at other places maybe a type trait like IsValidComponentType could be useful ?

Reply:
It is made during the component add (so only non decayed types can be searched, if you search for a non decayed type it will simply tell you it doesnt have it). In this part it is probably not necessary, following the same logic (trying to remove something that is not there)